core/math/math_funcs.h

```diff
@@ -377,51 +388,47 @@ class Math {
 		/* Formula from Wikipedia article on Bezier curves. */
 		double omt = (1.0 - p_t);
 		double omt2 = omt * omt;
-		double omt3 = omt2 * omt;
 		double t2 = p_t * p_t;
 		double t3 = t2 * p_t;
 
-		return p_start * omt3 + p_control_1 * omt2 * p_t * 3.0 + p_control_2 * omt * t2 * 3.0 + p_end * t3;
+		return omt * (p_start * omt2 + 3.0 * p_control_2 * t2) + 3.0 * p_control_1 * omt2 * p_t + p_end * t3;
 	}
 
 	static _ALWAYS_INLINE_ float bezier_interpolate(float p_start, float p_control_1, float p_control_2, float p_end, float p_t) {
 		/* Formula from Wikipedia article on Bezier curves. */
 		float omt = (1.0f - p_t);
 		float omt2 = omt * omt;
-		float omt3 = omt2 * omt;
 		float t2 = p_t * p_t;
 		float t3 = t2 * p_t;
 
-		return p_start * omt3 + p_control_1 * omt2 * p_t * 3.0f + p_control_2 * omt * t2 * 3.0f + p_end * t3;
+		return omt * (p_start * omt2 + 3.0f * p_control_2 * t2) + 3.0f * p_control_1 * omt2 * p_t + p_end * t3;
 	}
 
 	static _ALWAYS_INLINE_ double bezier_derivative(double p_start, double p_control_1, double p_control_2, double p_end, double p_t) {
```

Comment:
Looks like we are missing tests for this. Do you mind adding tests for this along with this PR to ensure that we don't introduce a regression